errors/error.go

```diff
@@ -116,13 +119,23 @@ func E(args ...interface{}) error {
 		case hcl.Range:
 			e.FileRange = arg
 		case hcl.Diagnostics:
-			diags = arg
+			diags := arg
 			if diags.HasErrors() {
```

Comment:
I just realized that `HasErrors()` looks for error severity, then if there are also Warnings just getting the diags[0] will get the wrong error. Not related to your change but have to be fixed.